radsat-sk/operation/crypt/encrypt.h

```diff
@@ -0,0 +1,9 @@
+#ifndef ENCRYPT_H_
```

Comment:
1. Why are you double declaring some of these functions? Are some of them meant to be private (only used internally)? If so, don't put them here in the header file
2. Why do you have a decrypt and encrypt file? Why not just have one file that combines both functionalities, since they are very similar and likely want to use the same internal functions (like padding calculations)?
3. Can you change the API so that there are **only** TWO public functions: `uint16_t aesEncrypt(uint8_t* inputBuffer, uint16_t bufferLength, uint8_t* encryptedBuffer)` and `uint16_t aesDecrypt(uint8_t* encryptedBuffer, uint16_t bufferLength, uint8_t* decryptedBuffer)` -> where both return the final size of the buffer (and 0 indicates failure)

Just an idea :)

Reply:
These header files should probably remain untouched, all the functions in there will have to be public. Maybe I'm just a C noob but as far as I can tell with the nature of pointers in C I don't think these functions can be nested? Because functions returning pointers doesn't really work the way you think it would? Don't really remember the full reasoning as I wrote these a long time ago but I think the calling process for encryption will have to be: 

paddedBuffSize = getPaddedBufSize(plainTextLen)
paddedBuff = buffer[paddedBuffSize]         (or do a malloc here to initialize the variable, whatever)
padBuffer(buff, paddeedBuff, buffLen, paddedBuffLen)
encrypt(paddedBuff,)

Then at the end of this process the paddedBuff buffer will contain the encrypted data. Kind of ugly to have to use three function calls but it's passable I think.